src/System/IO/Streams/SequenceId.hs

```diff
@@ -49,26 +55,39 @@ sequenceIdInputStream initSeqId getSeqId seqIdFaultHandler inStream =
 -- Example:
 --
 -- @
--- (outStream', getSeqId) <- 'sequenceIdOutputStream' 1 outStream
--- return $ 'System.IO.Streams.Combinators.contramapM' (addSeqId getSeqId) outStream'
+-- ghci> import qualified System.IO.Streams as Streams
+-- ghci> (os, getList) <- Streams.listOutputStream :: IO (OutputStream (Int,Int), IO [(Int,Int)])
+-- ghci> (outStream', getSeqId) <- sequenceIdOutputStream 0 (\seqId a -> (seqId, a)) os
+-- ghci> Streams.write (Just 6) outStream'
+-- ghci> Streams.write (Just 7) outStream'
+-- ghci> getList
+-- [(1,6),(2,7)]
+-- ghci> getSeqId
+-- 2
+-- ghci> Streams.write (Just 6) outStream'
+-- ghci> Streams.write (Just 7) outStream'
+-- ghci> getList
+-- [(1,6),(2,7)]
 -- @
 sequenceIdOutputStream :: Integral s
-                       => s                   -- ^ Initial sequence ID
-                       -> (s -> a -> b)       -- ^ Transformation function
-                       -> OutputStream b      -- ^ 'System.IO.Streams.OutputStream' to count the elements of
-                       -> IO (OutputStream a) -- ^ ('IO' 'SequenceId') is the action to run to get the current sequence ID
+                       => s                         -- ^ Initial sequence ID
+                       -> (s -> a -> b)             -- ^ Transformation function
+                       -> OutputStream b            -- ^ 'System.IO.Streams.OutputStream' to count the elements of
+                       -> IO (OutputStream a, IO s) -- ^ returns a new stream as well as an IO action to fetch and reset the updated seed value
```

Comment:
I'm a little be confused about the return of this function. Does it return the old sequence id value when you reset it or the new one? You should make that clear in the documentation. "fetch and reset" is a little ambiguous to me.